spyder/widgets/projects/tests/test_projectdialog.py

```diff
@@ -28,5 +29,29 @@ def test_project_dialog(qtbot):
     assert dlg
 
 
+@patch('spyder.widgets.projects.projectdialog.getexistingdirectory')
+def test_projectdialog_location(mock_getexistingdirectory):
+    """Run getexistingdirectory with mock values"""
```

Comment:
You can see the use of the monkeypatch fixture [here](https://github.com/spyder-ide/spyder/blob/5abd2a27e1c30b9ab4656e5aa290e99ac15b5ccb/spyder/widgets/variableexplorer/tests/test_dataframeeditor.py#L401-L410) to achieve what you proposed here :)